OK, too coarse at present as warnings are treated as errors, so the warnings for this workflow cause the failure:
```
2023-03-08T16:54:42.0508370Z [1;31mWARNING: A patch subject line should describe the change not the tool that found it[0m
2023-03-08T16:54:42.0733104Z #4:
2023-03-08T16:54:42.0733688Z Subject: [PATCH] workflows: We all love checkpatch, so add it to the CI
2023-03-08T16:54:42.0771746Z ##[error]WARNING: A patch subject line should describe the change not the tool that found it
2023-03-08T16:54:42.0782180Z 
2023-03-08T16:54:42.0782189Z 
2023-03-08T16:54:42.0782865Z [1;31mWARNING: added, moved or deleted file(s), does MAINTAINERS need updating?[0m
2023-03-08T16:54:42.0790362Z #18:
2023-03-08T16:54:42.0790629Z new file mode 100644
2023-03-08T16:54:42.0795511Z ##[error]WARNING: added, moved or deleted file(s), does MAINTAINERS need updating?
2023-03-08T16:54:42.0797772Z 
2023-03-08T16:54:42.0797782Z 
2023-03-08T16:54:42.0802122Z [1mtotal: 0 errors, 2 warnings, 16 lines checked[0m
2023-03-08T16:54:42.0803857Z [1;31m>> Failure[0m
```
The other two patches pass, and actually the warning over the reference to checkpatch is invalid in this case (I do describe what the patch is doing).

These tests aren't blocking merge, so perhaps we want to have warnings flagged such that we can make the call on whether we agree or not.

--- a/.github/workflows/checkpatch.yml
+++ b/.github/workflows/checkpatch.yml
@@ -1,0 +1,18 @@
+name: Advisory checkpatch review
+on: [pull_request]
+
+jobs:
+  review:
+    name: checkpatch review
+    runs-on: ubuntu-latest
+    steps:
+    - name: 'Calculate PR commits + 1'
+      run: echo "PR_FETCH_DEPTH=$(( ${{ github.event.pull_request.commits }} + 1 ))" >> $GITHUB_ENV
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: ${{ env.PR_FETCH_DEPTH }}
+    - name: Copy checkpatch.conf
+      run: cp ${{github.workspace}}/.github/workflows/ci_checkpatch.conf ${{github.workspace}}/.checkpatch.conf
+    - name: Run checkpatch review
+      uses: webispy/checkpatch-action@v9

--- a/.github/workflows/ci_checkpatch.conf
+++ b/.github/workflows/ci_checkpatch.conf
@@ -1,0 +1,4 @@
+--no-tree
+--ignore FILE_PATH_CHANGES
+--ignore GIT_COMMIT_ID
+--ignore SPDX_LICENSE_TAG

